src/main.rs

```diff
@@ -163,22 +159,37 @@ impl State {
         connection: &mut Connection,
         queue: &EventQueue<Self>,
     ) -> Result<()> {
-        // Setup OpenGL symbol loader.
-        unsafe {
-            egl_ffi::make_sure_egl_is_loaded()?;
-            gl::load_with(|symbol| egl::get_proc_address(symbol));
-        }
+        let mut wayland_display = WaylandDisplayHandle::empty();
+        wayland_display.display = connection.display().id().as_ptr() as *mut _;
+        let raw_display_handle = RawDisplayHandle::Wayland(wayland_display);
+
+        // Setup the OpenGL window.
+        let gl_display = unsafe { Display::new(raw_display_handle)? };
+
+        let template = ConfigTemplateBuilder::new()
+            .with_alpha_size(8)
+            .with_stencil_size(0)
+            .with_depth_size(0)
+            .build();
+
+        let egl_config = unsafe { gl_display.find_configs(template)?.next().unwrap() };
```

Comment:
Can this fail?

Reply:
The `?` -> yes. The other would mean that you don't have EGL configs matches what you request.

Reply:
I'm not sure how you should handle the other part, maybe some "No suitable EGL configuration found".

Reply:
Yeah since that can fail, I'd just throw in an `except` rather than unwrap. We don't need to be able to recover, but if it can realistically fail we should print an error that helps troubleshooting.